# Wikipedia Notable Life Expectancies

# [Notebook 3 of 4: Data Cleaning](https://github.com/teresahanak/wikipedia-life-expectancy/blob/main/wp_life_expect_data_clean2_thanak_2022_06_17.ipynb)

## Context

The


## Objective

The

### Data Dictionary

- Feature: Description

## Importing Necessary Libraries

In [1]:
# To structure code automatically
%load_ext nb_black

# To import/export sqlite databases
import sqlite3 as sql

# To help with reading and manipulating data
import pandas as pd
import numpy as np
import re

# To define maximum number of columns to be displayed in a dataframe
pd.set_option("display.max_columns", None)
# To define the maximum number of rows to be displayed in a dataframe
pd.set_option("display.max_rows", 200)

# To supress warnings
# import warnings

# warnings.filterwarnings("ignore")

# To set some visualization attributes
pd.set_option("max_colwidth", 150)

<IPython.core.display.Javascript object>

## Data Overview


### Reading, Sampling, and Checking Data Shape

In [2]:
# Reading the dataset
conn = sql.connect("wp_life_expect_clean1.db")
data = pd.read_sql("SELECT * FROM wp_life_expect_clean1", conn)

# Making a working copy
df = data.copy()

# Checking the shape
print(f"There are {df.shape[0]} rows and {df.shape[1]} columns.")

# Checking first 2 rows of the data
df.head(2)

There are 132584 rows and 20 columns.


day              name                                                 info  \
0   1  William Chappell  , 86, British dancer, ballet designer and director.   
1   1    Raymond Crotty         , 68, Irish economist, writer, and academic.   

                                                      link num_references  \
0  https://en.wikipedia.org/wiki/William_Chappell_(dancer)             21   
1             https://en.wikipedia.org/wiki/Raymond_Crotty             12   

   year    month info_parenth info_1           info_2  \
0  1994  January         None          British dancer   
1  1994  January         None         Irish economist   

                         info_3        info_4 info_5 info_6 info_7 info_8  \
0  ballet designer and director          None   None   None   None   None   
1                        writer  and academic   None   None   None   None   

  info_9 info_10 info_11   age  
0   None    None    None  86.0  
1   None    None    None  68.0

<IPython.core.display.Javascript object>

In [3]:
# Checking last 2 rows of the data
df.tail(2)

day           name  \
132582   9  Oleg Moliboga   
132583   9       Zou Jing   

                                                                         info  \
132582          , 69, Russian volleyball player, Olympic champion  and coach.   
132583  , 86, Chinese engineer, member of the Chinese Academy of Engineering.   

                                                     link num_references  \
132582        https://en.wikipedia.org/wiki/Oleg_Moliboga              2   
132583  https://en.wikipedia.org/wiki/Zou_Jing_(engineer)              3   

        year month info_parenth info_1                     info_2  \
132582  2022  June       (1980)         Russian volleyball player   
132583  2022  June         None                  Chinese engineer   

                                              info_3 info_4 info_5 info_6  \
132582                   Olympic champion  and coach   None   None   None   
132583  member of the Chinese Academy of Engineering   None   None   None   

       info_7 info_8 info_9 info_10 info_11   age  
132582   None   None   None    None    None  69.0  
132583   None   None   None    None    None  86.0

<IPython.core.display.Javascript object>

In [4]:
# Checking a sample of the data
df.sample(5)

day            name  \
55774   1   Ossie Hibbert   
89013  23  Viorel Morariu   
7119   23   Harold Hughes   
92734  16           Tu An   
30406   2  Bernard Loomis   

                                                                                                   info  \
55774                                                            , 62, Jamaican musician, heart attack.   
89013         , 85, Romanian rugby union player, Vernon Pugh Award for Distinguished Service recipient.   
7119                                                                         , 74, American politician.   
92734                                                                , 94, Chinese poet and translator.   
30406  , 82, American toymaker responsible for Strawberry Shortcake and  action figures, heart disease.   

                                               link num_references  year  \
55774   https://en.wikipedia.org/wiki/Ossie_Hibbert              4  2012   
89013  https://en.wikipedia.org/wiki/Viorel_Morariu              8  2017   
7119    https://en.wikipedia.org/wiki/Harold_Hughes             10  1996   
92734           https://en.wikipedia.org/wiki/Tu_An              9  2017   
30406  https://en.wikipedia.org/wiki/Bernard_Loomis              2  2006   

          month info_parenth info_1  \
55774      July         None          
89013       May         None          
7119    October         None          
92734  December         None          
30406      June         None          

                                                                           info_2  \
55774                                                           Jamaican musician   
89013                                                 Romanian rugby union player   
7119                                                          American politician   
92734                                                 Chinese poet and translator   
30406  American toymaker responsible for Strawberry Shortcake and  action figures   

                                                      info_3 info_4 info_5  \
55774                                           heart attack   None   None   
89013  Vernon Pugh Award for Distinguished Service recipient   None   None   
7119                                                    None   None   None   
92734                                                   None   None   None   
30406                                          heart disease   None   None   

      info_6 info_7 info_8 info_9 info_10 info_11   age  
55774   None   None   None   None    None    None  62.0  
89013   None   None   None   None    None    None  85.0  
7119    None   None   None   None    None    None  74.0  
92734   None   None   None   None    None    None  94.0  
30406   None   None   None   None    None    None  82.0

<IPython.core.display.Javascript object>

#### Observations:
- There are currently 132,584 rows and 20 columns.

### Checking Data Types, Duplicates, and Null Values

In [5]:
# Checking data types and null values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132584 entries, 0 to 132583
Data columns (total 20 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   day             132584 non-null  object 
 1   name            132584 non-null  object 
 2   info            132584 non-null  object 
 3   link            132584 non-null  object 
 4   num_references  132584 non-null  object 
 5   year            132584 non-null  int64  
 6   month           132584 non-null  object 
 7   info_parenth    49924 non-null   object 
 8   info_1          132584 non-null  object 
 9   info_2          132555 non-null  object 
 10  info_3          62620 non-null   object 
 11  info_4          12587 non-null   object 
 12  info_5          1505 non-null    object 
 13  info_6          217 non-null     object 
 14  info_7          33 non-null      object 
 15  info_8          7 non-null       object 
 16  info_9          1 non-null       object 
 17  info_10   

<IPython.core.display.Javascript object>

#### Observations:
- Our dataset was saved to and read from the database without any hiccups.
- Picking up where we left off, we will aim to track down the remaining missing values for `age`, starting with searching for digits in `info_2`.

### Remaining Missing Values for Age

In [6]:
# Checking number of remaining missing values
print(f'There are {df["age"].isna().sum()} missing values for age.')

There are 218 missing values for age.


<IPython.core.display.Javascript object>

#### Function to Save Indices of Rows Matching Regular Expressions Pattern to a List and Print Number of Rows with Match

In [7]:
# Define a function that takes dataframe, column name, and re pattern as arguments and returns list of indices
# for which column value matches re pattern
def rows_with_pattern(dataframe, column, pattern):
    """
    Takes input of dataframe, column name, and re pattern 
    and returns list of indices for rows that contain match
    for pattern anywhere within value for given column.
    
    dataframe: dataframe
    column: column name
    pattern: re pattern
    """
    index_list = []

    for i in dataframe.index:
        item = dataframe.loc[i, column]
        match = re.search(pattern, item)
        if match:
            index_list.append(i)
    print(
        f"There are {len(index_list)} rows with matching pattern in column '{column}'."
    )
    return index_list

<IPython.core.display.Javascript object>

#### Function to Use rows_with_pattern Function for Multiple Regular Expression Patterns

In [8]:
# Define a function that calls rows_with_pattern function for multiple re patterns
# returning a single list of indices for all rows with any pattern match


def multiple_patterns(dataframe, column, patterns):
    """
    Takes input dataframe, column, and list of re patterns and returns single list 
    of indices for rows in which a match for any pattern is found with re.search
    
    dataframe: dataframe
    column: column name
    patterns: list of re patterns
    """
    rows_combined = []

    # For loop to check each pattern
    for pattern in patterns:

        # List and number of rows matching each pattern
        print(pattern)
        rows_to_check = rows_with_pattern(dataframe, column, pattern)
        print("")

        # Add list for each pattern to combined list
        rows_combined += rows_to_check

    return rows_combined

<IPython.core.display.Javascript object>

### `info_2`

#### Rows Missing `age` with Digits in `info_2`

In [9]:
# Pattern for re
pattern = r"\d"

# List and number of rows matching pattern
rows_to_check = rows_with_pattern(
    df[(df["age"].isna()) & (df["info_2"].notna())], "info_2", pattern
)

# Examining the rows directly
df.loc[rows_to_check, :]

There are 47 rows with matching pattern in column 'info_2'.


day                            name  \
6846    17                     Spiro Agnew   
12861   14                     Muslimgauze   
14994   15               Željko Ražnatović   
15063   28                  Sarah Caudwell   
16112   30                   Max Showalter   
17793   15                     Joey Ramone   
18171    4                        Dipendra   
19018   28          Mohammad Khalequzzaman   
19118   12  Lord Hailsham of St Marylebone   
23753    6                     Jules Engel   
24883    8                 Muhammad Zaidan   
25174   30         Kioumars Saberi Foumani   
25202    6                  Kjell Hallbing   
26021   29               Shimon Wincelberg   
26821   28                  Karen Lancaume   
26980   16                  Narriman Sadek   
26998   19                 Kihachi Okamoto   
27026   22                      Lee Eun-ju   
27319    2               Pope John Paul II   
27728   30               Takanohana Kenshi   
28378   28                 Esther Szekeres   
28767   22                           Arman   
28963   15                Agenore Incrocci   
30076   20                  Stanley Hiller   
31239   15                    Guy François   
32149    9                     Cocoa Samoa   
43769   12              Saleban Olad Roble   
61936    1                     Basil Soper   
86953    8              Mohamud Muse Hersi   
109917  25                 Robert Levinson   
110688  12              Khalif Mumin Tohow   
110741  14                   Danny Delaney   
111354   2                   Justa Barrios   
111887  21              Kamrun Nahar Putul   
112736  23    Jean-Michel Bokamba-Yangouma   
113336  16             Cornelius Mwalwanda   
115273   2                       Fadma Abi   
121406  18           Michael Bedford-Jones   
121528  21                       Johny Lal   
122152  10           Salem Chandrasekharan   
122473  20                U. Visweswar Rao   
122746  30                    Venkat Subha   
123294  21                     Haribhushan   
124619   8        Fernando López de Olmedo   
126199   6                 Andrianafidisoa   
126893   2                     Irene Lalji   
127283  17                R. N. R. Manohar   

                                                                                                              info  \
6846                                , American politician, 77, 39th Vice President of the United States, leukemia.   
12861                                                       , , 37, British electronic musician, fungal infection.   
14994                                                             , , 47, Serbian mobster and paramilitary leader.   
15063                                                , , 60, British detective story writer and barrister, cancer.   
16112                                                   , , 83, American actor, composer, pianist, singer, cancer.   
17793                                            , , 49, American musician, lead singer for The Ramones, lymphoma.   
18171                                                                                , King of Nepal, 29, suicide.   
19018   , member of the then National Assembly of Pakistan and Union Minister of Labor, died in 28 September 2001.   
19118                                                                       , , 94, British lawyer and politician.   
23753                                     , Jules Engel, 94, American filmmaker, visual artist, and film director.   
24883                                      , , 55, Palestinian nationalist, founder of Palestine Liberation Front.   
25174                                                                                    , , 62, Iranian satirist.   
25202                                                                            , , 69, Norwegian Western author.   
26021                                                                          , , 80, American television writer.   
26821                       

<IPython.core.display.Javascript object>

#### Observations:
- There are several ages as sole integer year values and two as year ranges with two integers.
- The remaining entries are missing age, but do have digits, so order of processing matters here.
- We can safely remove any of these rows that contains a letter of the alphabet.

In [10]:
# Pattern for re
pattern = r"[a-z,A-Z]"

# List and number of rows matching pattern
rows_to_drop = rows_with_pattern(df.loc[rows_to_check, :], "info_2", pattern)

# Checking the rows
df.loc[rows_to_drop, :]

There are 21 rows with matching pattern in column 'info_2'.


day                          name  \
19018   28        Mohammad Khalequzzaman   
31239   15                  Guy François   
43769   12            Saleban Olad Roble   
109917  25               Robert Levinson   
110688  12            Khalif Mumin Tohow   
110741  14                 Danny Delaney   
111354   2                 Justa Barrios   
111887  21            Kamrun Nahar Putul   
112736  23  Jean-Michel Bokamba-Yangouma   
113336  16           Cornelius Mwalwanda   
115273   2                     Fadma Abi   
121406  18         Michael Bedford-Jones   
121528  21                     Johny Lal   
122152  10         Salem Chandrasekharan   
122473  20              U. Visweswar Rao   
122746  30                  Venkat Subha   
123294  21                   Haribhushan   
124619   8      Fernando López de Olmedo   
126199   6               Andrianafidisoa   
126893   2                   Irene Lalji   
127283  17              R. N. R. Manohar   

                                                                                                              info  \
19018   , member of the then National Assembly of Pakistan and Union Minister of Labor, died in 28 September 2001.   
31239                                       , Haitian Army colonel, participated in failed coups in 1989 and 2001.   
43769                            , Somali government minister, injuries sustained in the 2009 Shamo Hotel bombing.   
109917                                                       , American intelligence officer, missing since 2007.    
110688                                                   , Somali justice minister of Hirshabelle State, COVID-19.   
110741                                                                       , Irish Gaelic footballer , COVID-19.   
111354                                                 , American home care worker and labor organizer, COVID-19.    
111887                                                                         , Bangladeshi politician, COVID-19.   
112736                                                        , Congolese trade unionist and politician, COVID-19.   
113336                                                              , Malawian economist and politician, COVID-19.   
115273                                                                 , Moroccan surgeon and professor, COVID-19.   
121406                                                                      , Canadian suffragan bishop, COVID-19.   
121528                                                                        , Indian cinematographer , COVID-19.   
122152                                                                          , Indian film producer , COVID-19.   
122473                                                         , Indian film director  and screenwriter, COVID-19.   
122746                                                                                  , Indian actor , COVID-19.   
123294                                                                , Indian politician and guerrilla, COVID-19.   
124619                                                            , Spanish general commander of Ceuta , COVID-19.   
126199         , Malagasy military officer and convicted coupist, leader of the 2006 Malagasy coup d'état attempt.   
126893                                                     , Surinamese lawyer and television presenter, COVID-19.   
127283                                                                          , Indian film director , COVID-19.   

                                                               link  \
19018          https://en.wikipedia.org/wiki/Mohammad_Khalequzzaman   
31239     https://en.wikipedia.org/wiki/Guy_Fran%C3%A7ois_(colonel)   
43769              https://en.wikipedia.org/wiki/Saleban_Olad_Roble   
109917                https://en.wikipedia.org/wiki/Robert_Levinson   
110688             https://en.wikipedia.org/wiki/Khalif_Mumin_Tohow   
110741         

<IPython.core.display.Javascript object>

#### Observations:
- We can drop these rows, as they are missing the data for age.
- Extraction of age for single integer value and two integer ranges can follow.

#### Dropping Additional Rows with Age Data Absent

In [11]:
# Dropping rows, resetting index, and checking new shape of df
df.drop(rows_to_drop, inplace=True)
df.reset_index(inplace=True, drop=True)
df.shape

(132563, 20)

<IPython.core.display.Javascript object>

#### Remaining Rows with `age` values in `info_2`

In [12]:
# Regular expression for parenthesis and its contents
pattern = r"\d"

# Finding indices of rows that have pattern
rows_to_check = rows_with_pattern(
    df[(df["age"].isna()) & (df["info_2"].notna())], "info_2", pattern
)

# Checking unique values
df.loc[rows_to_check, :]["info_2"].unique()

There are 26 rows with matching pattern in column 'info_2'.


array(['77', '37', '47', '60', '83', '49', '29', '94', '55', '62', '69',
       '80', '32', '70', '81', '24', '84', '95', '76', '86', '61',
       '74–75', '79–80'], dtype=object)

<IPython.core.display.Javascript object>

#### Extracting `age` for Ranges with Two Values

In [13]:
# Pattern for re
pattern = r"(\d{1,3})(-|–|/| or )(\d{1,3})"

# List and number of rows matching pattern
rows_to_check = rows_with_pattern(
    df[(df["age"].isna()) & (df["info_2"].notna())], "info_2", pattern
)

# Checking sample of rows
df.loc[rows_to_check, :].sample(2)

There are 2 rows with matching pattern in column 'info_2'.


day                name  \
86950   8  Mohamud Muse Hersi   
61933   1         Basil Soper   

                                                      info  \
86950  , Somali politician, 79–80, President of Puntland .   
61933                              , British actor, 74–75.   

                                                   link num_references  year  \
86950  https://en.wikipedia.org/wiki/Mohamud_Muse_Hersi             12  2017   
61933         https://en.wikipedia.org/wiki/Basil_Soper              0  2013   

          month info_parenth             info_1 info_2                 info_3  \
86950  February  (2005–2009)  Somali politician  79–80  President of Puntland   
61933      June         None      British actor  74–75                   None   

      info_4 info_5 info_6 info_7 info_8 info_9 info_10 info_11  age  
86950   None   None   None   None   None   None    None    None  NaN  
61933   None   None   None   None   None   None    None    None  NaN

<IPython.core.display.Javascript object>

In [14]:
# For loop to find rows with values and pattern and calculate and extract age to age column and remove age from info_2
for i in df.loc[rows_to_check, :].index:
    item = df.loc[i, "info_2"]
    match = re.search(pattern, item)
    if match:
        age = (int(match.group(1)) + int(match.group(3))) / 2
        df.loc[i, "age"] = age
        df.loc[i, "info_2"] = re.sub(pattern, "", df.loc[i, "info_2"])

# Checking example rows
pd.concat([df[df["name"] == "Mohamud Muse Hersi"], df[df["name"] == "Basil Soper"]])

day                name  \
86950   8  Mohamud Muse Hersi   
61933   1         Basil Soper   

                                                      info  \
86950  , Somali politician, 79–80, President of Puntland .   
61933                              , British actor, 74–75.   

                                                   link num_references  year  \
86950  https://en.wikipedia.org/wiki/Mohamud_Muse_Hersi             12  2017   
61933         https://en.wikipedia.org/wiki/Basil_Soper              0  2013   

          month info_parenth             info_1 info_2                 info_3  \
86950  February  (2005–2009)  Somali politician         President of Puntland   
61933      June         None      British actor                          None   

      info_4 info_5 info_6 info_7 info_8 info_9 info_10 info_11   age  
86950   None   None   None   None   None   None    None    None  79.5  
61933   None   None   None   None   None   None    None    None  74.5

<IPython.core.display.Javascript object>

#### Extracting `age` as Single Integer

In [15]:
# List of patterns for age formats with single integer for age
pattern = r"\b(\d{1,3})\b"

# List and number of rows matching patterns
rows_to_check = rows_with_pattern(
    df[(df["age"].isna()) & (df["info_2"].notna())], "info_2", pattern
)

There are 24 rows with matching pattern in column 'info_2'.


<IPython.core.display.Javascript object>

In [17]:
# For loop to extract age pattern to age column
for i in df.loc[rows_to_check, :].index:
    item = df.loc[i, "info_2"]
    match = re.search(pattern, item)
    if match:
        age = int(match.group(1))
        df.loc[i, "age"] = age
        df.loc[i, "info_2"] = re.sub(pattern, "", df.loc[i, "info_2"])

# Re-checking number of rows matching patterns
rows_to_check = rows_with_pattern(
    df[(df["age"].isna()) & (df["info_2"].notna())], "info_2", pattern
)

There are 0 rows with matching pattern in column 'info_2'.


<IPython.core.display.Javascript object>

In [21]:
# Checking number of remaining missing values
print(f'There are {df["age"].isna().sum()} missing values for age.')
df[df["age"].isna()].sample(5)

There are 171 missing values for age.


day                           name  \
113214  12             Mohamed Abdi Hashi   
85111    8                    Aide Ganasi   
98059   11  Cheikhna Ould Mohamed Laghdaf   
108620   5                Mohammad Shafiq   
62642    9                Bhagwati Prasad   

                                                                               info  \
113214                                 , Somali politician, President of Puntland .   
85111                            , Papua New Guinean politician, MP , heart attack.   
98059                     , Mauritanian diplomat and politician, Foreign Minister .   
108620                                , Pakistani politician, MLA , cardiac arrest.   
62642   , Indian politician, Uttar Pradesh MLA for Ikauna , multiple organ failure.   

                                                               link  \
113214             https://en.wikipedia.org/wiki/Mohamed_Abdi_Hashi   
85111                     https://en.wikipedia.org/wiki/Aide_Ganasi   
98059   https://en.wikipedia.org/wiki/Cheikhna_Ould_Mohamed_Laghdaf   
108620   https://en.wikipedia.org/wiki/Mohammad_Shafiq_(politician)   
62642    https://en.wikipedia.org/wiki/Bhagwati_Prasad_(politician)   

       num_references  year      month            info_parenth  \
113214              6  2020       July             (2004–2005)   
85111              14  2016   November            (since 2012)   
98059               1  2018  September  (1962–1963, 1978–1979)   
108620              4  2020   February            (since 2015)   
62642               1  2013       July  (1967–1969, 1969–1974)   

                                     info_1                        info_2  \
113214                    Somali politician         President of Puntland   
85111          Papua New Guinean politician                            MP   
98059   Mauritanian diplomat and politician              Foreign Minister   
108620                 Pakistani politician                           MLA   
62642                     Indian politician  Uttar Pradesh MLA for Ikauna   

                        info_3 info_4 info_5 info_6 info_7 info_8 info_9  \
113214                    None   None   None   None   None   None   None   
85111             heart attack   None   None   None   None   None   None   
98059                     None   None   None   None   None   None   None   
108620          cardiac arrest   None   None   None   None   None   None   
62642   multiple organ failure   None   None   None   None   None   None   

       info_10 info_11  age  
113214    None    None  NaN  
85111     None    None  NaN  
98059     None    None  NaN  
108620    None    None  NaN  
62642     None    None  NaN

<IPython.core.display.Javascript object>

In [19]:
df[df["age"].isna()]

day                            name  \
13511   29                Faustin Birindwa   
22856   10                      Little Eva   
26314   18                   Alfred Maseng   
31766   21                Sir Harold Young   
32722   17                Joseph C. Casdin   
36487    2                      Justin Yak   
40580   20                   Nguyễn Bá Cẩn   
41574   22               Iftikhar Ali Khan   
44835    1                  Dragan Kujović   
45013   15               Gabriel Bien-Aimé   
49922   10       Abdi Shakur Sheikh Hassan   
51282   30            Va'aelua Eti Alesana   
56542   18                Hisham al-Hayali   
57149   26               Pape Alioune Diop   
57250    1                   Chronox Manek   
57557   20                Mataiasi Ragigia   
58497   10                     Harry Iauko   
59608   31               Mohammad Mahseiri   
61821   25                   Uday Mudliyar   
62215   16                     Daya Perera   
62642    9                 Bhagwati Prasad   
64111    6          Abdul Nasser Bani Hani   
66153   25                      Eric Green   
66826   24                Hamid Nawaz Khan   
68295    9       Flossie Gomile-Chidyaonga   
69558   15                      Adhu Awiti   
70090   13                     Hani Abbadi   
70878   23                    Mullah Ghani   
71418   22                 Barry McSweeney   
73798   20                 Nael al-Ajlouni   
73802   20                 Khalaf Masa'deh   
76207    1                  Nasser al-Hafi   
76595   22                   Martin Storey   
78269    1            Abdikarim Yusuf Adam   
78879    6            Jaafar Mohammed Saad   
79800   23                  Pablo Contessi   
80251   15               Muhayadin Mohamed   
81407   11                 A. R. Surendran   
81619   22                     Soran Singh   
82694   22                 Mohammad Kilani   
85111    8                     Aide Ganasi   
85146   11              Jum Jainudin Akbar   
86586   23                     Kang Ki-sop   
86611   24                  Lualemaga Faoa   
87442    3           Cyprian Ojwang Omollo   
89584   25       Agha Shahbaz Khan Durrani   
90190   28              Mamy Rakotoarivelo   
90698   27                 Martin Azonhiho   
90720   28                     Gulzar Khan   
92643   12                      Pat O'Rawe   
92858   23                      James Ligo   
93485   19                    Lutho Tapela   
94883   24                   Michael Voigt   
96631   24                 César Alvarenga   
96996   14                    Mick Langley   
97118   22             Ikramullah Gandapur   
97388    4                 Stephen Goodson   
97484   10               Emmanuel C. Aguma   
98026    9                Simon Adut Yuang   
98059   11   Cheikhna Ould Mohamed Laghdaf   
100118   2                  Kalipada Bauri   
100890   1            Ehsan-ul-Haq Piracha   
101368  21           Brahmanapalli Balaiah   
101386  21                       Lee Ocran   
102352   8               Sheikh Abdul Aziz   
102653  23                   Della Godfrey   
103901  22               Ambachew Mekonnen   
103902  22                 Se'are Mekonnen   
104163   5                       Mohan Das   
104454  19                     Dixon Seeto   
104796   5               Rashiram Debbarma   
105244  29                 Mohammad Muslim   
106357  27           Abul-Hasan al-Muhajir   
106505   3                Harish Mahapatra   
106688  13                      André Moes   
106955  26                    Gulzar Ahmed   
107278  11                     Albert Toro   
107567  25                        Niju Ram   
107786   3            M. Sakthivel Murugan   
107789   3                Rameshwar Prasad   
108512   1               Admiral K. Sangma   
108532   2       Kommareddy Surender Reddy   
108620   5                 Mohammad Shafiq   
108842  14           K. P. Rajendra Prasad   
109129  25                    Satya Nandan   
110578  10                Ceybil Jefferies   
11128

<IPython.core.display.Javascript object>